### PR TITLE
Enhance menu widget item class rendering

### PR DESCRIPTION
**Pull request by Kolyunya**
| Q             | A
| ------------- | ---
| Is bugfix?    | no
| New feature?  | yes
| Breaks BC?    | no
| Tests pass?   | yes
| Fixed issues  | #13770

### CODE REVIEW

File: framework/widgets/Menu.php [hunk 1]
```diff
@@ -210,9 +210,9 @@ protected function renderItems($items)
             }
             if (!empty($class)) {
                 if (empty($options['class'])) {
-                    $options['class'] = implode(' ', $class);
```
Comment (Kolyunya) [inline framework/widgets/Menu.php]: `$class` is going to be imploded by `renderTagAttributes()`.

File: framework/widgets/Menu.php [hunk 2]
```diff
@@ -210,9 +210,9 @@ protected function renderItems($items)
...
+                    $options['class'] = $class;
                 } else {
-                    $options['class'] .= ' ' . implode(' ', $class);
```
Comment (Kolyunya) [inline framework/widgets/Menu.php]: `$options['class']` is an array so it's not concatenated to `$class` but is merged with it.

File: tests/framework/widgets/MenuTest.php [hunk 3]
```diff
@@ -196,6 +196,91 @@ public function testActiveItemClosure()
...
+                    'label'  => 'item2',
+                    'url'    => '#',
+                    'options' => [
+                        'class' => [
+                            'another-class',
+                            'other--class',
+                            'two classes',
+                        ],
+                    ],
+                ],
```
Comment (Kolyunya) [inline tests/framework/widgets/MenuTest.php]: A test for the case when `$class` is empty.

File: framework/widgets/Menu.php [hunk 4]
```diff
@@ -210,9 +210,12 @@ protected function renderItems($items)
...
+                    if (!is_array($options['class'])) {
+                        $options['class'] = [$options['class']];
```
Comment (Kolyunya) [inline framework/widgets/Menu.php]: BC support. Probably a subject for removal.

File: framework/widgets/Menu.php [hunk 5]
```diff
@@ -210,9 +210,12 @@ protected function renderItems($items)
...
+                    }
+                    $options['class'] = ArrayHelper::merge($class, $options['class']);
```
Comment (Kolyunya) [inline framework/widgets/Menu.php]: `$options['class']` is an array so it's not concatenated to `$class` but is merged with it.

File: tests/framework/widgets/MenuTest.php [hunk 6]
```diff
@@ -196,6 +196,121 @@ public function testActiveItemClosure()
...
+            ]
+        ]);
+
+        $expected = <<<HTML
+<ul><li class="item--active someclass"><a href="#">item1</a></li>
+<li class="another-class other--class two classes"><a href="#">item2</a></li></ul>
+HTML;
+        $this->assertEqualsWithoutLE($expected, $output);
+
+        $output = Menu::widget([
```
Comment (Kolyunya) [inline tests/framework/widgets/MenuTest.php]: BC test.

File: framework/widgets/Menu.php [hunk 7]
```diff
@@ -210,9 +210,12 @@ protected function renderItems($items)
             }
             if (!empty($class)) {
                 if (empty($options['class'])) {
-                    $options['class'] = implode(' ', $class);
+                    $options['class'] = $class;
                 } else {
-                    $options['class'] .= ' ' . implode(' ', $class);
+                    if (!is_array($options['class'])) {
```
Comment (samdark) [inline framework/widgets/Menu.php]: Could be done as `$options['class'] = (array) $options['class'];`.

Comment (samdark) [inline framework/widgets/Menu.php]: Or even `$options['class'] = ArrayHelper::merge($class, (array) $options['class']);`.

Comment (Kolyunya) [inline framework/widgets/Menu.php]: I feel like the latter is an over complication of the merge statement. I used the former approach and added a comment clarifying why this is needed. What do you think?

**Review by samdark — CHANGES_REQUESTED** [review]
Code itself looks good. I'd break test method into three separate ones named after what they're testing.

**Comment by Kolyunya** [comment]
Refactored tests.

**Comment by samdark** [comment]
https://en.wikipedia.org/wiki/Pivot_table

File: framework/widgets/Menu.php (hunk 7 above)
Comment (samdark) [inline framework/widgets/Menu.php]: It's OK.

**Review by samdark — APPROVED** [review]

File: framework/widgets/Menu.php (hunk 1 above)
Comment (cebe) [inline framework/widgets/Menu.php]: there is `Html::addCssClass()` which could be used here as far as I see.

**Review by dynasource — APPROVED** [review]

File: framework/widgets/Menu.php (hunk 1 above)
Comment (Kolyunya) [inline framework/widgets/Menu.php]: Thank. That simplifies things up. Also altered resulting class order in tests.

**Review by SilverFire — APPROVED** [review]

**Comment by samdark** [comment]
Merged. Thanks!

**Comment by Kolyunya** [comment]
Glad to help!

**Pull request merged**